src/storm/mesos/MesosNimbus.java

```diff
@@ -496,49 +508,76 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
-                              .setRole(cpuRole))
-                          .addResources(Resource.newBuilder()
-                              .setName("mem")
-                              .setType(Type.SCALAR)
-                              .setScalar(Scalar.newBuilder().setValue(executorMem))
-                              .setRole(memRole))
-                  )
+                      .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+                      .setData(ByteString.copyFromUtf8(executorDataStr))
+                      .setName("storm supervisor executor")
+                      .setContainer(ContainerInfo.newBuilder()
+                          .setType(ContainerInfo.Type.DOCKER)
+                          .setDocker(ContainerInfo.DockerInfo.newBuilder()
+                              .setImage("mesos-storm")
+                              //using --pid=host so that we can generate pids visible to slave
+                              .addParameters(Parameter.newBuilder()
+                                  .setKey("pid")
+                                  .setValue("host")
+                                  .build())))
+                      .setCommand(CommandInfo.newBuilder()
+                          //don't need to download storm.yaml, it is bundled with the docker image
+                          //.addUris(CommandInfo.URI.newBuilder()
+                          //       .setValue(configUri))
+                          .setShell(false)
+                          .addArguments("supervisor")
+                          .addArguments("storm.mesos.MesosSupervisor")
+                          .addArguments("-c")
+                          .addArguments("mesos.master.url=" + masterUrl)
+                          .addArguments("-c")
+                          .addArguments("nimbus.host=" + nimbusHost)
+                          .addArguments("-c")
+                          .addArguments("storm.local.hostname=" + slaveHostname)
+                          .addAllArguments(getStormOptions()))
+                      .addResources(Resource.newBuilder()
+                          .setName("cpus")
+                          .setType(Type.SCALAR)
+                          .setScalar(Scalar.newBuilder().setValue(executorCpu))
+                          .setRole(cpuRole))
+                      .addResources(Resource.newBuilder()
+                          .setName("mem")
+                          .setType(Type.SCALAR)
+                          .setScalar(Scalar.newBuilder().setValue(executorMem))
+                          .setRole(memRole)))
                   .addResources(Resource.newBuilder()
-                      .setName("cpus")
-                      .setType(Type.SCALAR)
-                      .setScalar(Scalar.newBuilder().setValue(workerCpu))
-                      .setRole(cpuRole))
+                                    .setName("cpus")
```

Comment:
ugh sorry didn't realize the formatting was clobbered during this merge. will fix it